packages/expo-location/ios/EXLocation/TaskConsumers/EXLocationTaskConsumer.m

```diff
@@ -0,0 +1,100 @@
+// Copyright 2018-present 650 Industries. All rights reserved.
+
+#import <CoreLocation/CLLocationManager.h>
+
+#import <EXLocation/EXLocation.h>
+#import <EXLocation/EXLocationTaskConsumer.h>
+#import <EXTaskManagerInterface/EXTaskInterface.h>
+
+@interface EXLocationTaskConsumer ()
+
+@property (nonatomic, strong) CLLocationManager *locationManager;
+@property (nonatomic, strong) NSDictionary *options;
+
+@end
+
+@implementation EXLocationTaskConsumer
+
+- (void)dealloc
+{
+  [self reset];
+}
+
+# pragma mark - EXTaskConsumerInterface
+
+- (NSString *)taskType
+{
+  return @"location";
+}
+
+- (void)didRegisterTask:(id<EXTaskInterface>)task
+{
+  _task = task;
+  _locationManager = [CLLocationManager new];
```

Comment:
Are you sure it's executed on main thread? `CLLocationManager` likes to joke around when not initialized on main thread

Reply:
Yes, I'm sure. There are two ways to get here:
1. when you're calling register method from JS - then it's executed on the methodQueue from Location module (main queue).
2. when app is launched in the background - we're receiving that "notification" on the main queue and actually everything we do is being executed on the same queue.
Anyway, I'll check if I can simply put that into a block that is ensured to run on the main thread.